be/src/io/fs/file_reader.h

```diff
@@ -38,7 +40,11 @@ class FileReader {
     virtual const Path& path() const = 0;
 
     virtual size_t size() const = 0;
+
+    virtual bool closed() const = 0;
 };
 
+using FileReaderPtr = std::shared_ptr<FileReader>;
```

Comment:
if it is a shared ptr, FileReaderSPtr is better. If it is a unique ptr, FileReaderUPtr is better.